### ISSUE TITLE
Custom metadata types not included in metadata retrieved from a package using [-n PACKAGE-NAME]

### ISSUE THREAD

**Issue by simonwatson** [forcedotcom/cli#2299]
### Summary
When I use `project retrieve start` with the `-n` flag to retrieve the contents of a package, custom metadata types are not included in the retrieved ZIP file. 

### Steps To Reproduce
1. Create a 2GP package which contains custom metadata types (__mdt object, fields listed under customMetadata)
2. Install package into a scratch org
3. `sf project retrieve start -n "My package name"`

### Expected result
A zip file is retrieved with all metadata in the package (including custom metadata types)

### Actual result
A zip file is retrieved without custom metadata types in it

### System Information
@salesforce/cli/1.85.8 darwin-arm64 node-v16.8.0

**Comment by simonwatson** [comment]
Output from `sfdx version --verbose --json`:
```
{
  "cliVersion": "sfdx-cli/7.208.10",
  "architecture": "darwin-x64",
  "nodeVersion": "node-v18.15.0",
  "osVersion": "Darwin 22.3.0",
  "shell": "zsh",
  "rootPath": "/Users/simon/.local/share/sfdx/client/7.208.10-178b759",
  "pluginVersions": [
    "@muenzpraeger/sfdx-plugin 0.5.0 (user)",
    "@oclif/plugin-autocomplete 2.3.1 (core)",
    "@oclif/plugin-commands 2.2.16 (core)",
    "@oclif/plugin-help 5.2.10 (core)",
    "@oclif/plugin-not-found 2.3.26 (core)",
    "@oclif/plugin-plugins 3.1.5 (core)",
    "@oclif/plugin-search 0.0.17 (core)",
    "@oclif/plugin-update 3.1.21 (core)",
    "@oclif/plugin-version 1.3.5 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.40 (core)",
    "@oclif/plugin-which 2.2.22 (core)",
    "@salesforce/lwc-dev-server 2.11.0 (user)",
    "apex 2.3.3 (core)",
    "auth 2.8.4 (core)",
    "community 2.3.3 (core)",
    "custom-metadata 2.1.28 (core)",
    "data 2.4.2 (core)",
    "deploy-retrieve 1.13.3 (core)",
    "info 2.6.24 (core)",
    "limits 2.3.22 (core)",
    "org 2.9.17 (core)",
    "packaging 1.20.1 (core)",
    "schema 2.3.15 (core)",
    "settings 1.4.15 (core)",
    "signups 1.4.22 (core)",
    "source 2.10.19 (core)",
    "telemetry 2.2.1 (core)",
    "templates 55.5.0 (core)",
    "trust 2.4.24 (core)",
    "user 2.3.19 (core)",
    "sfdx-cli 7.208.10 (core)",
    "sfdx-package 0.2.0 (link) /Users/simon/git/axsy/sfdx-package",
    "sfdx-plugin-cli 1.2.0 (link) /Users/simon/git/axsy/sfdx-plugin-cli"
  ]
}
```

**Comment by simonwatson** [comment]
Output from `sfdx version --verbose --json:
```
{
  "cliVersion": "sfdx-cli/7.208.10",
  "architecture": "darwin-x64",
  "nodeVersion": "node-v18.15.0",
  "osVersion": "Darwin 22.3.0",
  "shell": "zsh",
  "rootPath": "/Users/simon/.local/share/sfdx/client/7.208.10-178b759",
  "pluginVersions": [
    "@muenzpraeger/sfdx-plugin 0.5.0 (user)",
    "@oclif/plugin-autocomplete 2.3.1 (core)",
    "@oclif/plugin-commands 2.2.16 (core)",
    "@oclif/plugin-help 5.2.10 (core)",
    "@oclif/plugin-not-found 2.3.26 (core)",
    "@oclif/plugin-plugins 3.1.5 (core)",
    "@oclif/plugin-search 0.0.17 (core)",
    "@oclif/plugin-update 3.1.21 (core)",
    "@oclif/plugin-version 1.3.5 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.40 (core)",
    "@oclif/plugin-which 2.2.22 (core)",
    "@salesforce/lwc-dev-server 2.11.0 (user)",
    "apex 2.3.3 (core)",
    "auth 2.8.4 (core)",
    "community 2.3.3 (core)",
    "custom-metadata 2.1.28 (core)",
    "data 2.4.2 (core)",
    "deploy-retrieve 1.13.3 (core)",
    "info 2.6.24 (core)",
    "limits 2.3.22 (core)",
    "org 2.9.17 (core)",
    "packaging 1.20.1 (core)",
    "schema 2.3.15 (core)",
    "settings 1.4.15 (core)",
    "signups 1.4.22 (core)",
    "source 2.10.19 (core)",
    "telemetry 2.2.1 (core)",
    "templates 55.5.0 (core)",
    "trust 2.4.24 (core)",
    "user 2.3.19 (core)",
    "sfdx-cli 7.208.10 (core)",
    "sfdx-package 0.2.0 (link) /Users/simon/git/axsy/sfdx-package",
    "sfdx-plugin-cli 1.2.0 (link) /Users/simon/git/axsy/sfdx-plugin-cli"
  ]
}
```

**Comment by iowillhoit** [comment]
Hello @simonwatson, sorry for the delay on this. Could you please try retrieving the same package using Workbench? If you are seeing the same result with Workbench, then it is not a CLI issue. A support case will need to be opened instead
